tests/test_core/test_alignment.py

```diff
@@ -2495,6 +2468,23 @@ def test_add_from_ref_aln(self):
             ValueError, aln1.add_from_ref_aln, aln2_wrong_refseq
         )  # test wrong_refseq
 
+    def test_to_moltype_annotations(self):
+        """correctly convert to specified moltype with proper sequence annotations"""
+        s1 = Sequence("TTTTTTAAAA", name="test_seq1")
+        s2 = Sequence("AAAATTTTTT", name="test_seq2")
+        s3 = Sequence("AATTTTTAAA", name="test_seq3")
+        s1.add_annotation(Feature, "exon", "fred", [(0, 6)])
+        s2.add_annotation(Feature, "exon", "fred", [(4, 10)])
+        s3.add_annotation(Feature, "exon", "fred", [(2, 7)])
+        data = {"seq1": s1, "seq2": s2, "seq3": s3}
+        seqs = self.Class(data=data)
+        seqs.add_feature("demo", "one", [(0, 1), (2, 4)])
+        rna = seqs.to_moltype("rna")
+        for seq in rna.seqs:
+            self.assertIsInstance(seq.data.annotations[0], _Annotatable)
+        self.assertIsInstance(seqs.annotations[0], _Annotatable)
```

Comment:
change this to compare the annotations between `aln` and `rna`: the number os the same, the coordinates are the same, the parent is consistent